services/webdav/pkg/service/v0/service.go

```diff
@@ -346,7 +346,11 @@ func (g Webdav) Thumbnail(w http.ResponseWriter, r *http.Request) {
 		switch e.Code {
 		case http.StatusNotFound:
 			// StatusNotFound is expected for unsupported files
-			renderError(w, r, errNotFound(notFoundMsg(tr.Filename)))
+			message := notFoundMsg(tr.Filename)
+			if strings.Contains(e.Detail, "could not find space") {
```

Comment:
Why do we want to change the error message only when the space is not found? Don't we need it also when a file is not found?

Reply:
For every 404 error, the error message was `File with name <filename> could not be located` which works for most of the cases.
But for a scenario where `user1` tries to get the file preview of another `user2`, the error message from oC10 server is `File not found: <filename> in 'user1'`.
Issue https://github.com/owncloud/ocis/issues/2071 is about different error message in ocis and there is this comment https://github.com/owncloud/ocis/issues/2071#issuecomment-847621421 to have same response like in oc10.

In this PR, the change in behavior is for `old/new` webdav endpoints, not for the spaces dav.

Reply:
So, my try here was to align error messages with oc10 behavior :smile:

Reply:
We can also change the test expectation if that's okay and close the issue

Reply:
Ah I see. But I'm afraid we can't do it like this. If we get an error message like `"could not find space"` it does NOT necessarily mean that a user tried to get a file preview of another user. There can be a dozen other reasons this error message appears.

Moreover I don't think adding such logic just to be compatible with oc10 will bring us any benefits.

Instead, if we really *must* to be compatible with oc10 in the error message, we should always return the oc10 error message instead of our own. However, since we do not have information about the space, the error messages will always look like: `File not found: lorem.txt in ''`

Reply:
> There can be a dozen other reasons this error message appears.
Yeah, I am also concerned about this.

It would be easier for the test to get out of the expected failure list if we could come up with some conclusion on the issue - whether to make it compatible or adapt the test expectation.

Reply:
Exactly what I meant :+1: We have two options
- drop oc10 compatibility and adjust test expectation
- always return error messages like `File not found: lorem.txt in ''`

If was to decide I would pick option 1. But I'm fine with the other option if I am outvoted.

Reply:
Option 2 also won't work for all the other test expectations.

Reply:
oc10 behavior is:
- Trying to get preview of non-existing file: error - `File with name <filename> could not be located`
- Trying to get preview of other user's file: error - `File not found: <filename> in 'user2'`

Reply:
updated the PR description if that helps to decide :sweat_smile:

Reply:
👍
If that poses the security risks then yeah security first.